function/src/main/proto/message.c

```diff
@@ -9,20 +9,39 @@ typedef struct _wrapper
 
 Wrapper filter(Wrapper input)
 {
-    SpringMessage *msg = spring_message__unpack(NULL, input.len, input.data);
-    SpringMessage *result = malloc(sizeof(SpringMessage));
-    spring_message__init(result);
-    result->payload = msg->payload;
-    result->headers = msg->headers;
-    result->n_headers = msg->n_headers;
-    int len = spring_message__get_packed_size(result);
+    Io__Cloudevents__V1__CloudEvent *msg = io__cloudevents__v1__cloud_event__unpack(NULL, input.len, input.data);
+    Io__Cloudevents__V1__CloudEvent *result = malloc(sizeof(Io__Cloudevents__V1__CloudEvent));
+    io__cloudevents__v1__cloud_event__init(result);
+
+    result->id = "1-5150"; //msg->id; TODO: see below - decorate by adding an attribute, not modify the id
+    result->source = msg->source;
+    result->spec_version = msg->spec_version;
+    result->type = msg->type;
+	result->data_case = msg->data_case;
+	result->binary_data = msg->binary_data;
+	result->text_data = msg->text_data;
+	result->proto_data = msg->proto_data;
+	/*
+	TODO: Figure out how to add another attribute 'decoratedby: "wasm"' (need to refamiliarize w/ '**' pointer-to-pointer usage.
```

Comment:
**I WANT TO** Decorate the passed in CE to "prove" the WASM did its thing. Had to get my head around the pointer-to-pointer, its been a while. :)  I ran out of time on it though. Will do this next time I sit down w/ this.

Reply:
You would need to `malloc` some memory for the target headers with space for the extra entry, and loop over the source headers, copying them (the pointers) into the target, instead of just assigning `target->headers = source->headers`. Then assign the last entry to your extra one.